src/Microsoft.AspNet.Mvc.ModelBinding/Binders/TypeSpecificModelBinder.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding
+{
+
+    public class TypeSpecificModelBinder : IModelBinder
+    {
+        public TypeSpecificModelBinder(Type modelType, IModelBinder modelBinder)
+        {
+            this.modelType = modelType;
```

Comment:
We do not use this. ... once you start the property name with capital letter it shouldn't be a problem. 
i.e ```ModelType = modelType```